doc/design/context.md

```diff
@@ -0,0 +1,166 @@
+## Context Design
+
+
+A Net is executed by single or several threads. A Context is related to a thread and records necessary runtime resources.
+
+Context is defined as follows:
+
+```
+class Context {};
```

Comment:
struct Context;
may be better, the `Context` is a lightweight data structure, users access its member data, without rich member functions needed. So a `struct` is enough.

reference [mxnet::Context](http://www.superjom.xyz/mxnetcode/codebrowser/include/mxnet/base.h.html#mxnet::Context)